### ISSUE TITLE
Bicycle Overlay: Distinguish between "signed bicycle allowed" and "not designated for cyclists"

### ISSUE THREAD

**Issue by westnordost** [streetcomplete/StreetComplete#4913]
Currently, the overlay only displays and allows to specify a foot-/cycle-/pathway as one of (tags in (...) are examples, more tags match the given categories):

- no cycling allowed (`bicycle=no`)
- path or trail, i.e. not designated for anyone in particular (`highway=path`)
- not designated for cyclists (`highway=footway`)
- shared-use path, i.e. designated to both pedestrians and cyclists, not segregated (`highway=cycleway + foot=designated` etc...)
- separated bike- and foot path (`highway=cycleway + foot=designated + segregated=yes` etc...)
- bike-only path (`highway=cycleway`)

The obvious omission here is "Cyclists are explicitly allowed on footway" (`highway=footway + bicycle=yes`).

There has been past discussions about this, the reason why it is not included is that it is not clear from these tags whether this is signed as such or the mapper was of the opinion that bikes are OK too. In the case that it isn't signed, it is not surveyable. In the worst case, StreetComplete users would think this tagging is wrong and must be corrected to "not designated for cyclists", removing the `bicycle=yes` when it indeed might be correct, due to some legislation or so that isn't signed.

The suggestion now is to use `bicycle:signed=yes` to denote that the access restriction (`bicycle=yes`) is indeed explicitly signed and thus surveyable.
`bicycle:signed=*` occured in related m-m-monster discussions in the forum several times (see https://community.openstreetmap.org/search?q=bicycle%3Asigned ) and as far as I remember, there was not really any opposition to it, only remarks about that it does not solve all the issues (discussed in that thread), e.g. that in Germany, if "bicycles allowed on footway" is signed, it means that bicycles must go at most at walking-speed.

I do no plan to implement this soon, I first wanted to drop this concrete suggestion here for comments.

**Comment by melbournefan** [comment]
In Australia (at least AU-VIC) 'Shared Paths' which are usually non segregated and intended for bicycles and foot traffic are signed at the beginning of the path and at the end of the path however they can have markings on the surface of the path, this is more prevalent now that Victoria is now making 'e-scooters' legal to use. Bicycles aren't allowed on footpaths `(footway=sidewalk)` unless the rider is under 12 years of age (though this varies in other states).

Bike only paths (or veloway) aren't common in Australia though the first one in Victoria is expected to open in the next few years as part of a larger road project. AFAIK there isn't a 'speed limit' to bicycles on 'shared paths' however there is a speed limit for e-scooters and that's at 20kph.

Overall, the ['Australian Tagging Guidelines'](https://wiki.openstreetmap.org/wiki/Australian_Tagging_Guidelines/Cycling_and_Foot_Paths#Sidewalk_Cycling) states that footpaths/sidewalks shouldn't have any bicycle specific tags unless it's a 'shared way', there are however footways with bicycle prohibited signs though those are mainly found in parks. I usually tag those as `bicycle=no`. Overall my opinion would be that if the path has a 'shared use' sign to tag it as `bicycle:signed=yes` other than that it's really up to the navigation app to ask users to observe local laws

Thanks

**Comment by mnalis** [comment]
> * shared-use path, i.e. designated to both pedestrians and cyclists, not segregated (highway=cycleway + foot=designated etc...)
> * separated bike- and foot path (highway=cycleway + foot=designated + segregated=yes etc...)
    bike-only path (highway=cycleway)

Ok, those two looks like **explicitly** marked bicycle ways (one not segregated, and other segregated), i.e. "explicitly" literally meaning `bicycle=designated` (like, authorities have _explicitly_ put a sign, drawn a pictogram etc) ?

> The obvious omission here is "Cyclists are explicitly allowed on footway" (highway=footway + bicycle=yes).

Ummm, did you mean _"Cyclists are **[implicitly](https://en.wiktionary.org/wiki/implicit)** allowed on footway"_ here? Like, there is some custom that implies that unless otherwise noted, any such and such surfaces designated for pedestrians may also be used by bicycles? But it is not written or signed anywhere?

> The suggestion now is to use bicycle:signed=yes to denote that the access restriction (bicycle=yes) is indeed explicitly signed and thus surveyable.

I'm not sure I follow. If there is a sign **explicitly** stating that bicycles may use some surface, than surely that surface is `bicycle=designated` (and `bicycle:signed=yes`, if you wish)?

Or do you wish to distinguish surfaces explicitly designated for bicycles **by sign** (i.e. `bicycle=designated`+`bicycle:signed=yes`) from situations where there is **no sign**, but bicycles are explicitly designated to use such surfaces **by law** (which would be `bicycle=designated`+`bicycle:signed=no`; e.g. if there was a law saying _"any designated footway wider than 5 meters is also designated for bicycle use"_)

**Comment by mnalis** [comment]
>   The obvious omission here is "Cyclists are explicitly allowed on footway" (highway=footway + bicycle=yes).

Or do you mean that there are two types of signs for bicycles from here which I see in german discussions https://wiki.openstreetmap.org/wiki/DE:Bicycle/Radverkehrsanlagen_kartieren :


- [traffic_sign](https://wiki.openstreetmap.org/wiki/DE:Key:traffic_sign)=DE:239,1022-10
  (which is supposed to be tagged with `foot=designated` + `bicycle=yes`), _and_

- [traffic_sign](https://wiki.openstreetmap.org/wiki/DE:Key:traffic_sign)=[DE:240](https://wiki.openstreetmap.org/wiki/DE:Tag:traffic_sign%3DDE:240)
  (which is supposed to be tagged with  `bicycle=designated` + `foot=designated` + `segregated=no`)


Because I fail to see the functional difference between them (apart from different signs) (and I do not trust automatic translators). To me they look like they mean the same: both pedestrians and cyclists may use this path, and there is no separation line drawn (i.e. they use it intermixed). 

And if there is **really no functional difference**, to me both should be **tagged the same** (like the latter). I mean, what is a difference between _"this is mixed footway and cycleway in one"_, _"this is a footway that cyclists may also use without restrictions"_, and _"this is a cycleway that pedestrians may also use without restrictions"_ ? :confused:

**Comment by matkoniecz** [comment]
>I'm not sure I follow. If there is a sign explicitly stating that bicycles may use some surface, than surely that surface is bicycle=designated (and bicycle:signed=yes, if you wish)?

In Poland there used to be a traffic sign combination ("pedestrian path/road + bicycles are exempt") that allowed cyclists to use sidewalk, without making it mandatory to use sidewalk. It was marked `bicycle=yes`

> Like, there is some custom that implies that unless otherwise noted, any such and such surfaces designated for pedestrians may also be used by bicycles? But it is not written or signed anywhere?

Yes, Poland has some cases like this (park rules allowing to use footways in a park but without traffic signs posted, though park rules may be posted , >=2m wide sidewalk next to road with >50km/h speed limit, paths on river embankments, many forest trails...)

**Comment by Helium314** [comment]
> Because I fail to see the functional difference between them

As far as I understand `1022-10` allows use with bicycle, while `DE:240` makes use obligatory. But tagging `bicycle=use_sidepath` on the correct road is not really something for SC.
Further, the combination `DE:239,1022-10` seems to limit bicycles to walking speed.

**Comment by westnordost** [comment]
> Or do you wish to distinguish surfaces explicitly designated for bicycles by sign (i.e. bicycle=designated+bicycle:signed=yes) from situations where there is no sign, but bicycles are explicitly designated to use such surfaces by law (which would be bicycle=designated+bicycle:signed=no; e.g. if there was a law saying "any designated footway wider than 5 meters is also designated for bicycle use")

Correct.

> Because I fail to see the functional difference between them

But there is a difference.

**Comment by frankie2784** [comment]
Could this be used in combination as bicycle=no and bicycle:signed=yes?

**Comment by westnordost** [comment]
It could, but would anyone tag `bicycle=no` if there is no explicit sign? The option "no cycling allowed" already exists.

**Comment by frankie2784** [comment]
Is "no cycling allowed" an option in SC? I don't see it

**Comment by frankie2784** [comment]
FWIW, bicycle:signed makes sense to me. My question would be what would the assumed default be in SC when there is no such tag?

**Comment by westnordost** [comment]
> In general I am trying to discuss new tagging schema on tagging mailing list / community.openstreetmap.org

Well, it was discussed on various threads on community.openstreetmap.org. I am not on the tagging mailing list. 

> bicycle:signed=* occured in related m-m-monster discussions in the forum several times (see https://community.openstreetmap.org/search?q=bicycle%3Asigned ) and as far as I remember, there was not really any opposition to it, only remarks about that it does not solve all the issues (discussed in that thread), e.g. that in Germany, if "bicycles allowed on footway" is signed, it means that bicycles must go at most at walking-speed.

**Comment by ownhardy** [comment]
I would be very much in favor of having this option in StreetComplete: To distinguish during my mapping activities between "path for pedestrians with no bike-related sign at all" and "path for pedestrians with a sign that explicitly allows bikes on this path". I am missing this feature very much.

**Comment by jabdoa2** [comment]
> I would be very much in favor of having this option in StreetComplete: To distinguish during my mapping activities between "path for pedestrians with no bike-related sign at all" and "path for pedestrians with a sign that explicitly allows bikes on this path". I am missing this feature very much.

I had exactly this case today with back and forth on the changeset. Luckily somebody corrected it for me to:
```
foot=designated
bicycle=yes
```
That should be the correct way for a pedestrian path where bikes are allowed (but its not a bicycle path).

**Comment by ownhardy** [comment]
(I hope this is not off topic:) If you get to improving the bicycle overlay, then I would also like to propose the following new feature:
Currently only a "path" can get all the different kind of bicycle tags described in your first comment.
But often there are also other types of "highways" that are officially labelled as bicycle and/or pedestrian way. For example in my city I regularly see things I would like to label as
- highway=service
  bicycle=designated
  foot=designated
  (often also segregated=no)
  (often including access=destination, but this is probably out of scope, here)
- highway=track
  bicycle=designated
  foot=designated
  (often also segregated=no)
  (often including access=agricultural, but this is probably out of scope, here)
I found a nice overview website, that presents many possible cycleway tagging options (including the two mentioned above) nicely: https://wiki.openstreetmap.org/wiki/DE:Bicycle/Radverkehrsanlagen_kartieren
Maybe this proposed feature could easily be implemented by just allowing the user of StreetComplete to use all the already existing bike tags not only on "path"s, but also on "track"s and "service" roads. 
This suggested feature would also keep StreetComplete users from changeing "highway=access" roads to "highway=path" roads, just to be able to tag (and see in the color coded map) properly that this way really is a bicycle road, as I have observed serveral times already. (Currently service ways that are correctly tagged as bicycle ways, are in StreetComplete displayed identical to non-bicycle ways in the bicycle overlay.)
Thank you for considering this.

**Comment by matkoniecz** [comment]
> "highway=access" roads

Something seems broken here. There is no such tag.

**Comment by ownhardy** [comment]
(Sorry for the typo, I meant "highway=service" in the last paragraph, as in the list above. I've now corrected my original post accordingly.)

**Comment by matkoniecz** [comment]
> would also keep StreetComplete users from changeing "highway=service" roads to "highway=path" 

Just to be clear: are they editing in external editor to make possible to later edit in SC?

**Comment by ownhardy** [comment]
( @matkoniecz: Yes, I assume that most StreetComplete users are also capable of using other OSM editors. In my post I ment changing the highway=* property in a different OSM editor in order to be able to tag and view the "nice" gapless bicycle path on the SC bicycle overlay afterwards. I feel tempted to do so as well, sometimes... But this seems to be a general effect of SC, since it gives acces only to a restricted subset of all OSM tags, people -probably including me- tend to only use the best fitting tag out of this subset in SC, to describe the reality they see outside, even if more specific and better fitting tags exist in the "full" OSM tag set. I see this as an advantage of SC in general, but it has side effects as well, for example this one described here.
If you like to discuss things further, I suggest that you start a thread in some forum and point me to this place, this bug report seems inappropriate to me for a longer general discussion.)

**Comment by tordans** [comment]
We ran into the lack of `highway=footway + bicycle=yes` in SC during our Mapathon events for https://wiki.openstreetmap.org/wiki/Brandenburg/Kampagne_Radnetz.

We would very much appreciate a solution for this! 

The issue is made more complex for us, since iD does not support sidewalk tagging very well on the centerline, so SC would be the first tool to actually support this kid of mapping in a guided way.

On the feature side, I can work with `bicycle<:side>:signed=yes`. I would very much prefer to have the proper `traffic_sign=*` tags added but I see that it is not the time to do this, yet.

However, the new system should look at the traffic_sign tags as a source for an internal `bicycle<:side>:signed=yes` indication. As in "when a way has a key traffic_sign and that key hold the values "DE:239" and "1022-10" than consider this `signed=yes`. This will make sure that more precise tagging can be used instead of the less precise generic tagging.

---

Side note: 

> I am surprised there is practically no feedback about this solution here.
Not sure what to take away from this - there is intense interest in discussing this topic to death, but no interest in a solution?

The fact that iD does not have great sidewalk tagging on the centerline or other tags that allow to detail the kind of cycle infrastructure puzzles we as well – but it looks to me to be the same lack of focus on a solution that you notice here.

I started experimenting with this for iD in https://github.com/openstreetmap/id-tagging-schema/pull/345 and similar takes but the current UI and schema does not support this kind of improvements well, yet.
